spec/compiler/apply_types_spec.cr

```diff
@@ -0,0 +1,580 @@
+require "./spec_helper"
+
+def run_source_typer_spec(input, expected_output,
```

Comment:
The specs are written with a before (INPUT) -> after (OUTPUT) pattern, which should hopefully showcase all of the use cases covered by the tool. All of the specs but the last one skip the `prelude` run for performance.